lucene/spatial3d/src/java/org/apache/lucene/spatial3d/geom/PlanetModel.java

```diff
@@ -118,71 +176,191 @@ public PlanetModel(final InputStream inputStream) throws IOException {
   }
 
   /** Find the maximum magnitude of all points on the ellipsoid.
    * @return the maximum magnitude for the planet.
    */
   public double getMaximumMagnitude() {
-    return Math.max(this.ab, this.c);
+    return Math.max(this.xyScaling, this.zScaling);
   }
   
   /** Find the minimum x value.
    *@return the minimum X value.
    */
   public double getMinimumXValue() {
-    return -this.ab;
+    return -this.xyScaling;
   }
   
   /** Find the maximum x value.
    *@return the maximum X value.
    */
   public double getMaximumXValue() {
-    return this.ab;
+    return this.xyScaling;
   }
 
   /** Find the minimum y value.
    *@return the minimum Y value.
    */
   public double getMinimumYValue() {
-    return -this.ab;
+    return -this.xyScaling;
   }
   
   /** Find the maximum y value.
    *@return the maximum Y value.
    */
   public double getMaximumYValue() {
-    return this.ab;
+    return this.xyScaling;
   }
   
   /** Find the minimum z value.
    *@return the minimum Z value.
    */
   public double getMinimumZValue() {
-    return -this.c;
+    return -this.zScaling;
   }
   
   /** Find the maximum z value.
    *@return the maximum Z value.
    */
   public double getMaximumZValue() {
-    return this.c;
+    return this.zScaling;
+  }
+
+  /** return the calculated mean radius (in meters) */
+  public double getMeanRadiusMeters() {
+    return this.r1;
+  }
+
+  /** encode the provided value from double to integer space */
+  public int encodeValue(double x) {
+    if (x > getMaximumMagnitude()) {
+      throw new IllegalArgumentException("value=" + x + " is out-of-bounds (greater than planetMax=" + getMaximumMagnitude() + ")");
+    }
+    if (x == getMaximumMagnitude()) {
+      x = Math.nextDown(x);
+    }
+    if (x < -getMaximumMagnitude()) {
+      throw new IllegalArgumentException("value=" + x + " is out-of-bounds (less than than -planetMax=" + -getMaximumMagnitude() + ")");
+    }
+    long result = (long) Math.floor(x / DECODE);
+    assert result >= Integer.MIN_VALUE;
+    assert result <= Integer.MAX_VALUE;
+    return (int) result;
+  }
+
+  /**
+   * Decodes a given integer back into the radian value according to the defined planet model
+   */
+  public double decodeValue(int x) {
+    double result;
+    if (x == MIN_ENCODED_VALUE) {
+      // We must special case this, because -MAX_VALUE is not guaranteed to land precisely at a floor value, and we don't ever want to
+      // return a value outside of the planet's range (I think?).  The max value is "safe" because we floor during encode:
+      result = -MAX_VALUE;
+    } else if (x == MAX_ENCODED_VALUE) {
+      result = MAX_VALUE;
+    } else {
+      // We decode to the center value; this keeps the encoding stable
+      result = (x+0.5) * DECODE;
+    }
+    assert result >= -MAX_VALUE && result <= MAX_VALUE;
+    return result;
+  }
+
+  /** Encode a provided GeoPoint into DocValue sortable integer space */
```

Comment:
I wonder if instead of adding all those methods, the planet model should just return the docValueEncoder?

Reply:
I added a getter for the docValueEncoder in PlanetModel